packages/teams-js/src/private/messageChannels.ts

```diff
@@ -0,0 +1,59 @@
+import { requestPortFromParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * Namespace to request message ports from the host application.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace messageChannels {
+  let telemetryPort: MessagePort | undefined;
+  /**
+   * @hidden
+   *
+   * Fetches a MessagePort to batch telemetry through the host's telemetry worker.
+   * @returns MessagePort.
+   *
+   * @throws Error if {@linkcode app.initialize} has not successfully completed,
+   * if the host does not support the feature, or if an error is thrown in message handling.
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export async function getTelemetryPort(): Promise<MessagePort> {
+    // If the port has already been initialized, return it.
+    if (telemetryPort) {
```

Comment:
Why telemetry port is returned before ensureInitialized call? Won't it be problematic in case of multiple calls to this function ?

Reply:
If there is a cached port inside the iFrame, it returns immediately. If not, it will call the parent to fetch the port, and store it for any future calls.

The cached port will never be present before the initialization is complete, because we cannot fetch it from the parent until after initialization is complete.

Can you explain further what flow you are envisioning causing issues?

Reply:
Discussed offline. As every app has it's own iFrame there won't be issues of accessing telemetry port from another iframe.